src/libraries/IterableMap.sol

```diff
@@ -0,0 +1,104 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+
+contract IterableMap {
+
+    struct Item {
+        bool hasItem;
+        uint256 value;
+        uint256 offset;
+    }
+
+    address owner;
+    bool initialized;
+    uint256[] itemsArray;
+    Item[] itemsMap;
+    uint numberOfItems;
```

Comment:
Solidity has proper arrays, which means we should no longer need `numberOfItems` as you can just do `itemsArray.length`.  Anywhere where we are adding an item and incrementing the length we can instead do:
```
itemsArray.push(newItem);
```
Anywhere where we are removing an item from the top of the array and shortening we can do:
```
delete itemsArray[itemsArray.length];
itemsArray.length -= 1;
```